buffer/src/main/java/io/netty/buffer/UnreleasableByteBuf.java

```diff
@@ -57,7 +57,9 @@ public ByteBuf readSlice(int length) {
 
     @Override
     public ByteBuf readRetainedSlice(int length) {
-        return new UnreleasableByteBuf(buf.readRetainedSlice(length));
+        ByteBuf retainedSlice = buf.readRetainedSlice(length);
```

Comment:
we have 2 options for each of these methods:

1. use `buf.readRetainedSlice` and decrement the reference count of the parent buffer after
2. use `buf.slice`

Option 1 has the advantage of potentially using pooled derived buffers, but must do extra reference counting.